### PR TITLE
implemented controllers: MovieController, MovieSessionController, Aut…

### PR DESCRIPTION
**Pull request by MaliukDaria**
…henticationController; added all necessary DTO; added get by id method into DAO layer

### CODE REVIEW

**Review by TimoxaV — APPROVED** [review]

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    @PostMapping
+    public void addMovieSession(@RequestBody MovieSessionRequestDto requestDto) {
+        MovieSession movieSession = movieSessionMapper.mapToMovieSession(requestDto);
+        movieSessionService.add(movieSession);
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAvailableMovieSessions(
+            @RequestParam Long movieId, @RequestParam String date) {
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controller/MovieSessionController.java]: can we replace String with LocalDate here?
- read about @ DateTimeFormat annotation in spring web

File: src/main/java/com/dev/cinema/dao/impl/GenericDaoImpl.java [hunk 2]
```diff
@@ -40,4 +42,14 @@ public T add(T item) {
...
     }
+
+    @Override
+    public Optional<T> get(Long id, Class<T> clazz) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<T> getByIdQuery = session.createQuery(
+                    "FROM " + clazz.getSimpleName() + " m WHERE m.id = :id", clazz)
+                    .setParameter("id", id);
+            return getByIdQuery.uniqueResultOptional();
+        }
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/GenericDaoImpl.java]: maybe better:
```
try (Session session = sessionFactory.openSession()) {
        return session.get(clazz, id);
}
```

File: src/main/java/com/dev/cinema/main/MainApp.java [hunk 3]
```diff
@@ -38,6 +38,7 @@ public static void main(String[] args) {
         theLordOfTheRings.setDescription("Lord");
         movieService.add(pulpFiction);
         movieService.add(theLordOfTheRings);
+        movieService.get(1L);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/main/MainApp.java]: we can remove class with main method

File: src/main/java/com/dev/cinema/model/dto/MovieSessionResponseDto.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.dev.cinema.model.dto;
+
+import java.time.LocalDateTime;
+import lombok.Data;
+
+@Data
+public class MovieSessionResponseDto {
+    private Long id;
+    private LocalDateTime showTime;
+    private String movieTitle;
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionResponseDto.java:11]: let's add movie id too

File: src/main/java/com/dev/cinema/main/MainApp.java (hunk 3 above)
Comment (MaliukDaria) [reply src/main/java/com/dev/cinema/main/MainApp.java]: done

File: src/main/java/com/dev/cinema/dao/impl/GenericDaoImpl.java (hunk 2 above)
Comment (MaliukDaria) [reply src/main/java/com/dev/cinema/dao/impl/GenericDaoImpl.java]: done

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java (hunk 1 above)
Comment (MaliukDaria) [reply src/main/java/com/dev/cinema/controller/MovieSessionController.java]: done

File: src/main/java/com/dev/cinema/model/dto/MovieSessionResponseDto.java (hunk 4 above)
Comment (MaliukDaria) [reply src/main/java/com/dev/cinema/model/dto/MovieSessionResponseDto.java:11]: done

File: src/main/java/com/dev/cinema/controller/AuthenticationController.java [hunk 5]
```diff
@@ -15,7 +16,8 @@ public AuthenticationController(AuthenticationService authenticationService) {
     }
 
     @PostMapping("/registration")
-    public void registerUser(@RequestBody UserRequestDto userRequestDto) {
-        authenticationService.register(userRequestDto.getEmail(), userRequestDto.getPassword());
+    public User registerUser(@RequestBody UserRequestDto userRequestDto) {
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controller/AuthenticationController.java]:
```suggestion
    public void registerUser(@RequestBody UserRequestDto userRequestDto) {
```
void is ok) later we will return status code

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        movieSessionService.add(movieSession);
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAvailableMovieSessions(
+            @RequestParam Long movieId,
+            @RequestParam @DateTimeFormat(pattern = "dd.MM.yyyy") LocalDate date) {
+        return movieSessionService.findAvailableSessions(
+                movieId, date)
+                .stream()
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controller/MovieSessionController.java]:
```suggestion
        return movieSessionService.findAvailableSessions(movieId, date).stream()
```

File: src/main/java/com/dev/cinema/controller/AuthenticationController.java (hunk 5 above)
Comment (MaliukDaria) [reply src/main/java/com/dev/cinema/controller/AuthenticationController.java]: ой, это я проверяла как @RequestBody и @ResponceBody работают в одном методе и забыла вернуть все обратно) Исправила

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java (hunk 6 above)
Comment (MaliukDaria) [reply src/main/java/com/dev/cinema/controller/MovieSessionController.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**